force-app/main/default/classes/NKS_KnowledgeBrokenLinksFinder.cls

```diff
@@ -0,0 +1,245 @@
+                                try {
+                                    HttpResponse res = calloutHttpReq(
+                                        req,
+                                        'callout:' +
+                                        link.replace('.', '_') +
+                                        str.substringAfter(link)
+                                    );
+                                    if (link.contains('.no')) {
+                                        statusCode = detectRedirect(req, res);
+                                    } else {
+                                        statusCode = res.getStatusCode();
+                                    }
+
+                                    if (statusCode != 200 && statusCode != 500) {
+                                        resultList.add(str);
+                                    }
+                                } catch (Exception ex) {
+                                    //System.debug('It is possbile this link is broken, check the link.');
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            if (resultList.size() > 0) {
+                finalList.put(article.Id, resultList);
+            }
+        }
+        System.debug('Number of articles with dead links: ' + finalList.size());
+        System.debug(finalList);
+    }
+
+    public void finish(Database.BatchableContext bc) {
+        // toAddress: nav.kontaktsenter.fag.og.kvalitet@nav.no
+        String[] toAddresses = new List<String>{ 'sara.mohammadi@nav.no' };
+        sendEmail(finalList, toAddresses);
+    }
+
+    @TestVisible
+    private static Integer detectRedirect(HttpRequest req, HttpResponse res) {
+        Boolean redirect;
+        Integer statusCode = res.getStatusCode();
+
+        if (Test.isRunningTest()) {
+            System.debug('Running tests.');
+            return null;
+        } else {
+            if (statusCode >= 300 && statusCode <= 307 && statusCode != 306) {
+                do {
+                    redirect = false; // reset the value each time
+                    String loc = res.getHeader('Location'); // get location of the redirect
+                    if (loc == null) {
+                        redirect = false;
+                        continue;
+                    }
+                    req = new HttpRequest();
+                    req.setEndpoint(loc);
+                    req.setMethod('GET');
+                    Http http = new Http();
+                    res = http.send(req);
+                    if (statusCode != 500) {
+                        // 500 = fail
+                        if (statusCode >= 300 && statusCode <= 307 && statusCode != 306) {
+                            redirect = true;
+                        }
+                    }
+                } while (redirect && Limits.getCallouts() != Limits.getLimitCallouts());
+            }
```

Comment:
Hvis vi har to linker som redirecter til broken links, vil ikke denne bare fange opp den første? Siden den andre kjøringen vil ha nådd callout limits